### PR TITLE
Add tokenization support for billing info

### PR DESCRIPTION
**Pull request by aaron-suarez**
Adds gateway_code, gateway_token, and card_type in billing info xml.

Playground script:

```C#
// Assumes account "x" exists on this site
// Assumes gateway_code "jhpqd551excn" is the code for vantiv on this site
using System;
using Recurly;

namespace TestRig
{
    class CreateBillingInfo
    {
        public static void Run(string[] args)
        {
            var account = Accounts.Get("x");
            var info = new BillingInfo(account);
            info.FirstName = "John";
            info.LastName = "Smith";
            info.Address1 = "123 Main St";
            info.City = "San Fransisco";
            info.State = "CA";
            info.Country = "US";
            info.PostalCode = "90212";
            info.GatewayToken = "4111115851581111";
            info.GatewayCode = "jhpqd551excn";
            info.CardType = BillingInfo.CreditCardType.Visa;
            info.ExpirationMonth = 10;
            info.ExpirationYear = 2020;

            try
            {
                info.Create();
            }
            catch(RecurlyException e)
            {
                Console.WriteLine(e.Errors[0]);
            }
        }
    }
}
```

### CODE REVIEW

File: Library/BillingInfo.cs [hunk 1]
```diff
@@ -401,6 +405,13 @@ internal override void WriteXml(XmlTextWriter xmlWriter)
                     xmlWriter.WriteElementString("external_hpp_type", ExternalHppType.Value.ToString().EnumNameToTransportCase());
 
                 }
+
+                if (!GatewayCode.IsNullOrEmpty())
+                {
+                    xmlWriter.WriteElementString("gateway_code", GatewayCode);
+                    xmlWriter.WriteElementString("gateway_token", GatewayToken);
+                    xmlWriter.WriteElementString("card_type", CardType.ToString().ToLower());
```
Comment (bhelx) [inline Library/BillingInfo.cs]: Will this produce the correct string for every value? You may need to use `EnumNameToTransportCase` here.

Comment (aaron-suarez) [reply Library/BillingInfo.cs]: Fixed

File: Library/BillingInfo.cs [hunk 2]
```diff
@@ -401,6 +405,15 @@ internal override void WriteXml(XmlTextWriter xmlWriter)
                     xmlWriter.WriteElementString("external_hpp_type", ExternalHppType.Value.ToString().EnumNameToTransportCase());
 
                 }
+
+                if (!GatewayCode.IsNullOrEmpty())
+                {
+                    xmlWriter.WriteElementString("gateway_code", GatewayCode);
+                    xmlWriter.WriteElementString("gateway_token", GatewayToken);
+                    var card = CardType.ToString().EnumNameToTransportCase();
```
Comment (bhelx) [inline Library/BillingInfo.cs]: Can we drop a comment here explaining why this is needed?

Comment (aaron-suarez) [reply Library/BillingInfo.cs]: Yup

**Pull request merged**